File: server-scala-play-server/app/api/TwilioApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.BigDecimal
import model.TwiMLResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class TwilioApiController @Inject()(cc: ControllerComponents, api: TwilioApi) extends AbstractController(cc) {
  /**
    * POST /api/:version/sms/buyoffer/:appKey?body=[value]&from=[value]&currencyType=[value]
    * @param appKey the application key
    */
  def smsBuyOffer(version: BigDecimal, appKey: String): Action[AnyContent] = Action { request =>
    def executeApi(): TwiMLResponse = {
      val body = request.getQueryString("Body")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("Body", "query string")
        }
        
      val from = request.getQueryString("From")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("From", "query string")
        }
        
      val currencyType = request.getQueryString("currencyType")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("currencyType", "query string")
        }
        
      api.smsBuyOffer(version, appKey, body, from, currencyType)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/TwilioApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.TwiMLResponse;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class TwilioApiController extends Controller {
    private final TwilioApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private TwilioApiController(Config configuration, TwilioApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result smsBuyOffer(Http.Request request, BigDecimal version,String appKey) throws Exception {
        String valuebody = request.getQueryString("Body");
        String body;
        if (valuebody != null) {
            body = valuebody;
        } else {
            throw new IllegalArgumentException("'Body' parameter is required");
        }
        String valuefrom = request.getQueryString("From");
        String from;
        if (valuefrom != null) {
            from = valuefrom;
        } else {
            throw new IllegalArgumentException("'From' parameter is required");
        }
        String valuecurrencyType = request.getQueryString("currencyType");
        String currencyType;
        if (valuecurrencyType != null) {
            currencyType = valuecurrencyType;
        } else {
            throw new IllegalArgumentException("'currencyType' parameter is required");
        }
        return imp.smsBuyOfferHttp(request, version, appKey, body, from, currencyType);
    }

}
